src/Publisher/FilesystemPublisher.php

```diff
@@ -174,8 +204,35 @@ protected function saveToPath($content, $filePath)
         // Move the temporary file to the desired location (prevents unlocked files from being read during write)
         $publishPath = $this->getDestPath() . DIRECTORY_SEPARATOR . $filePath;
         Filesystem::makeFolder(dirname($publishPath));
+        $successWithPublish = rename($temporaryPath, $publishPath);
+        if ($successWithPublish) {
+            if (FilesystemPublisher::config()->get('use_gzip_compression')) {
+                $publishPath = $this->compressFile($publishPath);
+            }
+        }
 
-        return rename($temporaryPath, $publishPath);
+        return file_exists($publishPath);
+    }
+
+    protected function compressFile(string $publishPath): string
+    {
+        // read original
+        $data = implode('', file($publishPath));
```

Comment:
This removes the new lines, doesn't it? perhaps `file_get_contents` would be better?